account_invoice_merge_massive/wizard/invoice_merge.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models, api
+
+
+class InvoiceMerge(models.TransientModel):
+    _inherit = "invoice.merge"
+
+    @api.model
+    def _dirty_check(self):
+        if self._context.get('active_model') == 'account.invoice':
+            group_invoices = self._group_invoices_to_treat()
+            for invoices in group_invoices:
+                my_context = self.env.context.copy()
+                my_context.update({'active_ids': invoices.ids})
+                res = super(InvoiceMerge,
+                            self.with_context(my_context))._dirty_check()
+        else:
+            res = super(InvoiceMerge, self)._dirty_check()
+        return res
+
+    @api.multi
+    def merge_invoices(self):
+        invoice_ids = []
+        group_invoices = self._group_invoices_to_treat()
+        for invoices in group_invoices:
+            my_context = self.env.context.copy()
+            my_context.update({'active_ids': invoices.ids})
+            res = super(InvoiceMerge,
+                        self.with_context(my_context)).merge_invoices()
+            domain = res.get('domain', [])
+            for inv_ids in domain:
+                for invoice_id in inv_ids[2]:
```

Comment:
for inv_ids in domain.filtered(lambda x: x[0] == 'id'):
    invoice_ids += inv_ids[2]

Reply:
@agaldona , ya he realiado la mejora de programacion, y he vuelto a subir el módulo. Eskerrik asko.